src/dsp_tools/commands/xmlupload/check_consistency_with_ontology.py

```diff
@@ -0,0 +1,123 @@
+# sourcery skip: use-fstring-for-concatenation
+import regex
+from lxml import etree
+from regex import Pattern
+
+from dsp_tools.commands.xmlupload.models.ontology_diagnose_models import InvalidOntologyElements, OntoCheckInformation
+from dsp_tools.commands.xmlupload.ontology_client import OntologyClientLive
+from dsp_tools.models.exceptions import BaseError
+
+defaultOntologyColon: Pattern[str] = regex.compile(r"^:\w+$")
+knoraUndeclared: Pattern[str] = regex.compile(r"^\w+$")
+genericPrefixedOntology: Pattern[str] = regex.compile(r"^[A-Za-z]+-?[A-Za-z]+:\w+$")
+
+
+def do_xml_consistency_check(onto_client: OntologyClientLive, root: etree._Element) -> None:
```

Comment:
I thought the entire point of having a protocol is that the protocol should be used here? 

```suggestion
def do_xml_consistency_check(onto_client: OntologyClient, root: etree._Element) -> None:
```

Reply:
you're right, I missed that